packages/rum/src/domain/record/serialization/serializeNode.ts

```diff
@@ -22,12 +22,21 @@ import type {
 } from '../../../types'
 import { NodeType } from '../../../types'
 import { getSerializedNodeId, getValidTagName, setSerializedNodeId } from './serializationUtils'
-import type { SerializeOptions } from './serialization.types'
+import type { SerializeOptions, SerializedNodeCache } from './serialization.types'
 import { serializeStyleSheets } from './serializeStyleSheets'
 import { serializeAttributes } from './serializeAttributes'
 
-export function serializeNodeWithId(node: Node, options: SerializeOptions): SerializedNodeWithId | null {
-  const serializedNode = serializeNode(node, options)
+export function serializeNodeWithId(
+  node: Node,
+  options: SerializeOptions,
+  cache?: SerializedNodeCache
+): SerializedNodeWithId | null {
+  const serializedNodeId = getSerializedNodeId(node)
+  if (serializedNodeId && cache && cache.has(serializedNodeId)) {
+    return cache.get(serializedNodeId) as SerializedNodeWithId
+  }
```

Comment:
**💭 thought:** ‏I don't think we should cache serialized nodes, but serialized stylesheets ([this function](https://github.com/DataDog/browser-sdk/blob/52ee5263cfdd87308c1083393bf9f3f7e194b4bc/packages/rum/src/domain/record/serialization/serializeStyleSheets.ts#L3)). The node has attributes that can change, and we don't invalidate the cache in that case. This might be trickier than expected.

Instead, the cache could be `Map<CSSStyleSheet, string>`

Reply:
As discussed we'll keep caching the serialized node but listening on mutation observer to invalidate cache.